src/software/ai/hl/stp/stp.cpp

```diff
@@ -118,12 +118,13 @@ std::vector<std::unique_ptr<Intent>> STP::getIntentsFromCurrentPlay(const World&
     std::vector<std::unique_ptr<Intent>> intents;
     if (current_tactics)
     {
-        assignRobotsToTactics(world, *current_tactics);
+        std::vector<std::shared_ptr<Tactic>> assigned_tactics =
+            assignRobotsToTactics(world, *current_tactics);
 
         ActionWorldParamsUpdateVisitor action_world_params_update_visitor(world);
         TacticWorldParamsUpdateVisitor tactic_world_params_update_visitor(world);
 
-        for (const std::shared_ptr<Tactic>& tactic : *current_tactics)
+        for (const std::shared_ptr<Tactic>& tactic : assigned_tactics)
         {
             tactic->accept(tactic_world_params_update_visitor);
```

Comment:
Note that we're using the return value of `assignRobotsToTactics` to update world parameters and get actions, so if a tactic gets deleted, then we won't get actions from it

Reply:
Good catch.